dbms/src/Functions/FunctionsJson.h

```diff
@@ -530,11 +612,671 @@ class FunctionsJsonArray : public IFunction
+        UInt8 from_type_code,
+        Int32 flen,
+        size_t size)
+    {
+        size_t reserve_size = (size * (1 + 1)) + data_from->getSizeForReserve();
+        JsonBinary::JsonBinaryWriteBuffer write_buffer(data_to, reserve_size);
+        for (size_t i = 0; i < size; ++i)
+        {
+            const auto & slice = data_from->getWhole();
+            if constexpr (is_binary_str)
+            {
+                if (unlikely(flen <= 0))
+                {
+                    JsonBinary::appendOpaque(
+                        write_buffer,
+                        JsonBinary::Opaque{from_type_code, StringRef{slice.data, slice.size}});
+                }
+                else
+                {
+                    auto size_t_flen = static_cast<size_t>(flen);
+                    if (slice.size >= size_t_flen)
+                    {
+                        JsonBinary::appendOpaque(
+                            write_buffer,
+                            JsonBinary::Opaque{from_type_code, StringRef{slice.data, size_t_flen}});
+                    }
+                    else
+                    {
+                        ColumnString::Chars_t buf;
+                        buf.resize_fill(size_t_flen, 0);
+                        std::memcpy(buf.data(), slice.data, slice.size);
+                        JsonBinary::appendOpaque(
+                            write_buffer,
+                            JsonBinary::Opaque{from_type_code, StringRef{buf.data(), size_t_flen}});
+                    }
+                }
+            }
+            else
+            {
+                JsonBinary::appendOpaque(
+                    write_buffer,
+                    JsonBinary::Opaque{from_type_code, StringRef{slice.data, slice.size}});
+            }
+            writeChar(0, write_buffer);
+            offsets_to[i] = write_buffer.count();
+            data_from->next();
+        }
+    }
+
+    template <bool is_nullable>
+    static void doExecuteForParsingJson(
+        ColumnString::Chars_t & data_to,
+        ColumnString::Offsets & offsets_to,
+        const std::unique_ptr<IStringSource> & data_from,
+        const NullMap & null_map_from,
+        size_t size)
+    {
+        size_t reserve_size = (size * (1 + 1)) + data_from->getSizeForReserve();
+        JsonBinary::JsonBinaryWriteBuffer write_buffer(data_to, reserve_size);
+        simdjson::dom::parser parser;
+        for (size_t i = 0; i < size; ++i)
+        {
+            if constexpr (is_nullable)
+            {
+                if (null_map_from[i])
+                {
+                    writeChar(0, write_buffer);
+                    offsets_to[i] = write_buffer.count();
+                    data_from->next();
+                    continue;
+                }
+            }
+
+            const auto & slice = data_from->getWhole();
+            if (unlikely(slice.size == 0))
+                throw Exception("Invalid JSON text: The document is empty.");
```

Comment:
Will `parser.parse()` not return error if size == 0?

Reply:
The parser will return an error, but in tidb `str.empty` and `invalid json text` has different error messages..